### PR TITLE
add wafv2_ip_set module

### PR DESCRIPTION
**Pull request by markuman**
##### SUMMARY
Add new module to handle wafv2 ip sets.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
wafv2_ip_set  
wafv2_ip_set_info

### CODE REVIEW

**Comment by markuman** [comment]
aws terminator needs rw access to wafv2

`botocore.exceptions.ClientError: An error occurred (AccessDeniedException) when calling the ListIPSets operation: User: arn:aws:sts::966509639900:assumed-role/ansible-core-ci-test-prod/prod=shippable=ansible-collections=community.aws=1701.19 is not authorized to perform: wafv2:ListIPSets on resource: arn:aws:wafv2:us-east-1:966509639900:regional/ipset/*`

**Comment by markuman** [comment]
There are failures that are not related with my PR

```
TASK [lambda : parallel lambda creation 1/4] ***********************************
```

File: plugins/modules/wafv2_ip_set.py
Comment (jillr) [inline plugins/modules/wafv2_ip_set.py:126]: Could you please wrap your boto client calls in try/except blocks that handle both ClientError and BotoCoreError, and exits with module.fail_json_aws? (that method will handle to_native on the exception message for you)

I don't see any use of waiters so we shouldn't need to import WaiterError

File: tests/integration/targets/wafv2_ip_set/tasks/main.yml
Comment (jillr) [inline tests/integration/targets/wafv2_ip_set/tasks/main.yml:12]: Is this here for the 2.9 tests?  You can open a PR directly against the stable-2.9 branch for mod_defaults, like this: https://github.com/ansible/ansible/pull/72145

File: tests/integration/targets/wafv2_ip_set/tasks/main.yml
Comment (jillr) [inline tests/integration/targets/wafv2_ip_set/tasks/main.yml:38]: Could you please assert the return values for the non-check_mode tests?

**Review by jillr — COMMENTED** [review]
This PR will need a changelog.

meta/runtime.yml is causing ci_complete to be run, if you have issues with the tests timing out or exceeding request limits give me a ping on irc and I can rerun them (a few are still failing now, but they're nothing to do with this change)

File: tests/integration/targets/wafv2_ip_set/tasks/main.yml
Comment (markuman) [reply tests/integration/targets/wafv2_ip_set/tasks/main.yml:12]: Hm, yes. Do we need to backport it to ansible 2.9?

File: tests/integration/targets/wafv2_ip_set/tasks/main.yml
Comment (markuman) [reply tests/integration/targets/wafv2_ip_set/tasks/main.yml:29]: Some are already there: https://github.com/ansible-collections/community.aws/pull/449/files#diff-5c4696663139c385c3dccf20dd0f8354b8685d88dfbed62bc170e0f32ece05abR132  
I've added more checks.

File: tests/integration/targets/wafv2_ip_set/tasks/main.yml
Comment (markuman) [reply tests/integration/targets/wafv2_ip_set/tasks/main.yml]: Ok I reverse the `&aws_connection_info` construct.  
Locally everything pass.

**Comment by jillr** [comment]
recheck

**Review by jillr — APPROVED** [review]
lgtm when tests are passing

**Comment by markuman** [comment]
recheck

**Comment by markuman** [comment]
> lgtm when tests are passing

there are still so many failures which are not related to this PR. How can we made them pass?

**Comment by markuman** [comment]
Required: https://github.com/ansible/ansible/pull/73975 for pass 2.9 integration tests

**Comment by s-hertel** [comment]
Thanks for putting so much work into these new modules, they look great :smiley: I just wanted to mention that once 2.9 becomes security-only, new modules shouldn't be too much of a hassle to test in CI with YAML anchors + module_defaults actions. It should just be an additional two lines for a new module, like 
```
module_defaults:
  group/aws: &aws_connection_info
    ...
  # Use the fqcn here to work regardless of how the plugin is called and ensure future compatibility
  community.aws.new_module:
    <<: *aws_connection_info
```

(But don't worry; #73975 will be included in the upcoming final non-only-security 2.9 cycle, it just hasn't been added yet because those get merged in batches by the release engineer)

**Comment by tremble** [comment]
@s-hertel what worries me about using the YAML Anchors is that this will mean that when someone forgets to update the group in runtime.yml the integration tests won't catch it.  This is a really poor experience for collection maintainers.

**Comment by tremble** [comment]
Test failures are due to known flakes.  WAF tests are running cleanly and taking around 10-20 seconds.

**Comment by tremble** [comment]
Thanks for your work on this @markuman sorry it's taken a while to get everything merged.

**Pull request merged**
